src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbs.tsx

```diff
@@ -175,38 +193,58 @@ class BreadcrumbsContainer extends React.Component<Props, State> {
     };
   };
 
-  handleFilter = (breadcrumbFilterGroups: BreadcrumbFilterGroups) => () => {
+  handleFilter = (filterGroups: BreadcrumbFilterGroups) => () => {
     //types
-    const breadcrumbFilterGroupTypes = breadcrumbFilterGroups.filter(
-      breadcrumbFilterGroup => breadcrumbFilterGroup.groupType === 'type'
-    );
-
-    // TODO(Priscila): implement levels
-    // const breadcrumbFilterGroupLevels = breadcrumbFilterGroups
-    //   .filter(breadcrumbFilterGroup => breadcrumbFilterGroup.groupType === 'level')
-    //   .map(breadcrumbFilterGroup => breadcrumbFilterGroup.type);
-
-    this.setState({
-      filteredBreadcrumbs: this.state.breadcrumbs.filter(breadcrumb => {
-        const foundBreadcrumbFilterData = breadcrumbFilterGroupTypes.find(
-          crumbFilterData => crumbFilterData.type === breadcrumb.type
+    const breadcrumbFilterGroupTypes = filterGroups
+      .filter(
+        breadcrumbFilterGroup =>
+          breadcrumbFilterGroup.groupType === 'type' && breadcrumbFilterGroup.isChecked
+      )
+      .map(breadcrumbFilterGroup => breadcrumbFilterGroup.type);
+
+    //levels
+    const breadcrumbFilterGroupLevels = filterGroups
+      .filter(
+        breadcrumbFilterGroup =>
+          breadcrumbFilterGroup.groupType === 'level' && breadcrumbFilterGroup.isChecked
+      )
+      .map(breadcrumbFilterGroup => breadcrumbFilterGroup.type);
+
+    const filteredByFilter = this.state.breadcrumbs.filter(({type, level}) => {
```

Comment:
If we add more filter types you might want to change this design and have each category of filter produce a function that can be used in a chain of functions to reduce the items.

```javascript
// psuedo code
const levelFilter = makeLevelFilter();
const typeFilter =makeTypeFilter();
const filters = [levelFilter, typeFilter];

const filteredCrumbs = this.state.breadcrumbs.filter(item => {
  return filters.map(filterFunc => filterFunc(item)).includes(true); 
});
```

Reply:
sure... I really like it..let me implement this solution in another PR ;-) Thank you!